modules/accounts/email.nix

```diff
@@ -6,6 +6,45 @@ let
 
   cfg = config.accounts.email;
 
+  gpgModule = types.submodule {
```

Comment:
For the GPG options I would suggest covering a useful set of options that are in the intersection of multiple mail programs. I.e., if a user sets an option here they should be reasonably sure that their enabled mail client will actually support the option.

Having a look at neomutt, alpine, alot, and Evolution I would suggest having just

- `gpg.key : types.str` - the key ID to use by default (e.g. `02391E86CEC4F00EABB301EAEC0FE921CB500C61`)

and possibly also

- `gpg.signByDefault : types.bool`
- `gpg.encryptByDefault : types.bool`

although I'm not so certain about these. But it would help to have some form of table having the programs and their features listed to guide the design.